Index: frontend/src/components/uikit/modal/useModal.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/frontend/src/components/uikit/modal/useModal.tsx b/frontend/src/components/uikit/modal/useModal.tsx
--- a/frontend/src/components/uikit/modal/useModal.tsx	(revision b0835be135ba5da91ac6d6802da1e01833492e09)
+++ b/frontend/src/components/uikit/modal/useModal.tsx	(revision 6e5e1e177daa026788223a2d6a597f5dfc7d91d8)
@@ -257,6 +257,7 @@
                   errorText={options.fieldError}
                   onKeyDown={(event) => {
                     if (event.key === 'Enter') {
+                      commonClose();
                       options.resolve(fieldValue);
                       setFieldValue('');
                     }
