Index: frontend/src/components/uikit/modal/useModal.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/frontend/src/components/uikit/modal/useModal.tsx b/frontend/src/components/uikit/modal/useModal.tsx
--- a/frontend/src/components/uikit/modal/useModal.tsx	(revision fb8f972d2e064d85c9153b2912f110ab46b2e745)
+++ b/frontend/src/components/uikit/modal/useModal.tsx	(revision 8264b0abc02473a16e2b1c0f3780e9c7a1f0c04a)
@@ -282,9 +282,10 @@
                   <Button
                     className={styles.button}
                     color={
-                      options.cancelButtonColor ?? options.okButtonColor
+                      options.cancelButtonColor ??
+                      (options.okButtonColor
                         ? ButtonColor.Primary
-                        : ButtonColor.Secondary
+                        : ButtonColor.Secondary)
                     }
                     title={options.cancelButtonText ?? i18n.t('cancel_button')}
                     onClick={onClose}
