Index: frontend/src/helpers/form/react-hook-form-helper.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/frontend/src/helpers/form/react-hook-form-helper.ts b/frontend/src/helpers/form/react-hook-form-helper.ts
--- a/frontend/src/helpers/form/react-hook-form-helper.ts	(revision c4f135936ea6157a73e00ef78ad3d1b2b4f16402)
+++ b/frontend/src/helpers/form/react-hook-form-helper.ts	(revision 751e17f20b3a0e4742c007961b0d947428f926c5)
@@ -27,7 +27,7 @@
         keepDirtyValues: true,
       });
     }
-  }, [defaultValues]);
+  }, [superjson.stringify(defaultValues)]);
 }
 
 export function requiredRule() {
