Index: frontend/vite.config.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/frontend/vite.config.ts b/frontend/vite.config.ts
--- a/frontend/vite.config.ts	(revision ef06db92ff199be94f13cf67e7f20c5f02380fc6)
+++ b/frontend/vite.config.ts	(revision c9372b49deb3a7b98bff032fc3225ed8af809913)
@@ -1,4 +1,5 @@
 import { defineConfig, loadEnv } from 'vite';
+import autoprefixer from 'autoprefixer';
 import tsconfigPaths from 'vite-tsconfig-paths';
 import react from '@vitejs/plugin-react';
 import reactSwc from '@vitejs/plugin-react-swc';
@@ -75,6 +76,9 @@
       modules: {
         localsConvention: 'camelCase',
       },
+      postcss: {
+        plugins: [autoprefixer],
+      },
     },
     build: {
       outDir: 'build',
