Index: webapi/src/MccSoft.TemplateApp.App/Program.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/webapi/src/MccSoft.TemplateApp.App/Program.cs b/webapi/src/MccSoft.TemplateApp.App/Program.cs
--- a/webapi/src/MccSoft.TemplateApp.App/Program.cs	(revision 947bfff69042f6d24b0ddc78f6809f25f87ceeba)
+++ b/webapi/src/MccSoft.TemplateApp.App/Program.cs	(revision eb5e0f391b801c29f3bf1dde786f28b45da8e62e)
@@ -9,10 +9,9 @@
 
 var builder = WebApplication.CreateBuilder(args);
 
-builder.Configuration.AddJsonFile("appsettings.local.json", true).AddEnvironmentVariables();
-
 if (!builder.Environment.IsEnvironment("Test"))
 {
+    builder.Configuration.AddJsonFile("appsettings.local.json", true).AddEnvironmentVariables();
     builder.Host.UseSerilog(
         (hostingContext, loggerConfiguration) =>
         {
