Index: webapi/src/MccSoft.TemplateApp.App/Setup/SetupSwagger.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/webapi/src/MccSoft.TemplateApp.App/Setup/SetupSwagger.cs b/webapi/src/MccSoft.TemplateApp.App/Setup/SetupSwagger.cs
--- a/webapi/src/MccSoft.TemplateApp.App/Setup/SetupSwagger.cs	(revision 92e0094a6688a59a5de4bb873a912b4330a31f19)
+++ b/webapi/src/MccSoft.TemplateApp.App/Setup/SetupSwagger.cs	(revision 25c5e261030237bcac274c1889d77c847e00bb6a)
@@ -1,11 +1,13 @@
 using System.Text.Json;
 using MccSoft.TemplateApp.App.Settings;
 using MccSoft.WebApi.Patching;
+using MccSoft.WebApi.Serialization.FromQueryJson;
 using MccSoft.WebApi.Swagger;
 using NJsonSchema;
 using NJsonSchema.Generation.TypeMappers;
 using NSwag;
 using NSwag.AspNetCore;
+using NSwag.Examples;
 using NSwag.Generation.AspNetCore;
 using NSwag.Generation.Processors.Security;
 
@@ -24,16 +26,20 @@
         var configuration = builder.Configuration;
 
         var swaggerOptions = configuration.GetSwaggerOptions();
-        services.AddOpenApiDocument(options =>
-        {
-            ConfigureOpenApiDocument(options, swaggerOptions);
-            options.SchemaSettings.SchemaProcessors.Add(
-                new RequireValueTypesSchemaProcessor(makePatchRequestFieldsNullable: false)
-            );
-            options.SchemaSettings.SchemaProcessors.Add(
-                new ValidationProblemDetailsSchemaProcessor()
-            );
-        });
+        services.AddOpenApiDocument(
+            (options, provider) =>
+            {
+                options.AddExamples(provider);
+                options.AddJsonQuerySupport();
+                ConfigureOpenApiDocument(options, swaggerOptions);
+                options.SchemaSettings.SchemaProcessors.Add(
+                    new RequireValueTypesSchemaProcessor(makePatchRequestFieldsNullable: false)
+                );
+                options.SchemaSettings.SchemaProcessors.Add(
+                    new ValidationProblemDetailsSchemaProcessor()
+                );
+            }
+        );
         services.AddOpenApiDocument(options =>
         {
             ConfigureOpenApiDocument(options, swaggerOptions);
@@ -73,7 +79,7 @@
                 Type = OpenApiSecuritySchemeType.ApiKey,
                 Name = "Authorization",
                 Description = "Copy 'Bearer ' + valid JWT token into field",
-                In = OpenApiSecurityApiKeyLocation.Header
+                In = OpenApiSecurityApiKeyLocation.Header,
             }
         );
 
