Index: webapi/src/MccSoft.TemplateApp.App/MccSoft.TemplateApp.App.csproj
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/webapi/src/MccSoft.TemplateApp.App/MccSoft.TemplateApp.App.csproj b/webapi/src/MccSoft.TemplateApp.App/MccSoft.TemplateApp.App.csproj
--- a/webapi/src/MccSoft.TemplateApp.App/MccSoft.TemplateApp.App.csproj	(revision a0fb32374d448a8147d30342cf55f34115d5e345)
+++ b/webapi/src/MccSoft.TemplateApp.App/MccSoft.TemplateApp.App.csproj	(revision 55ae1b03e164ef0741810de4395b2d65ccc7ee05)
@@ -104,6 +104,12 @@
     </Content>
     <Content Include="..\..\..\README.md">
       <Link>README.md</Link>
+    </Content>
+    <Content Include="..\..\Directory.Build.props">
+      <Link>Directory.Build.props</Link>
+    </Content>
+    <Content Include="..\..\Directory.Packages.props">
+      <Link>Directory.Packages.props</Link>
     </Content>
     <None Update="logs;idsrv4test.pfx;Dockerfile">
       <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
